okhttp/src/main/java/okhttp3/internal/http2/Http2Connection.kt

```diff
@@ -639,16 +635,22 @@ class Http2Connection internal constructor(builder: Builder) : Closeable {
           val newStream = Http2Stream(streamId, this@Http2Connection, false, inFinished, headers)
           lastGoodStreamId = streamId
           streams[streamId] = newStream
-          listenerExecutor.execute("OkHttp $connectionName stream $streamId") {
-            try {
-              listener.onStream(newStream)
-            } catch (e: IOException) {
-              Platform.get().log("Http2Connection.Listener failure for $connectionName", INFO, e)
-              ignoreIoExceptions {
-                newStream.close(ErrorCode.PROTOCOL_ERROR, e)
+
+          // Use a different task queue for each stream because they should be handled in parallel.
+          val taskName = "OkHttp $connectionName stream $streamId"
+          taskRunner.newQueue(taskName).schedule(object : Task(taskName) {
+            override fun runOnce(): Long {
+              try {
+                listener.onStream(newStream)
+              } catch (e: IOException) {
+                Platform.get().log("Http2Connection.Listener failure for $connectionName", INFO, e)
```

Comment:
Should this logging be a part of the task contract?  e.g. rethrow to log?

Reply:
or task.logErrors() fluent API?

Reply:
No I don’t think so. Uncaught exceptions should go to the VM’s UncaughtExceptionHandler because they indicate a bug.

This IOException is expected and recoverable and handled differently.